agent/eventhandler/task_handler.go

```diff
@@ -214,44 +228,53 @@ func (handler *TaskHandler) taskStateChangesToSend() []api.TaskStateChange {
 				Task:    task,
 			}
 			event.SetTaskTimestamps()
-
-			events = append(events, event)
+			events[taskARN] = event
 		}
 	}
-	return events
-}
 
-// batchContainerEventUnsafe collects container state change events for a given task arn
-func (handler *TaskHandler) batchContainerEventUnsafe(event api.ContainerStateChange) {
-	// if this is a managed agent state change, let's try to bundle it in an existing container event
-	if event.GetEventType() == statechange.ManagedAgentEvent {
-		containerChanges, ok := handler.tasksToContainerStates[event.TaskArn]
-		if ok {
-			for i, cs := range containerChanges {
-				if cs.ContainerName == event.ContainerName && cs.Status == event.Status {
-					cs.ManagedAgents = append(cs.ManagedAgents, event.ManagedAgents...)
-					containerChanges[i] = cs
-					return
-				}
-			}
+	for taskARN := range handler.tasksToManagedAgentStates {
+		if _, ok := events[taskARN]; ok {
+			continue
 		}
-	}
+		if task, ok := handler.state.TaskByArn(taskARN); ok {
+			// We do not allow the ticker to submit managed agent state updates for
```

Comment:
this comment is sooo confusing, but I get it a similar one was already there

Reply:
agree.  I'll leave it for now.  We can refactor comments as we refactor code.